flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/FlinkConfigBuilder.java

```diff
@@ -80,6 +83,17 @@ public FlinkConfigBuilder applyFlinkConfiguration() {
         return this;
     }
 
+    public FlinkConfigBuilder applyLogConfiguration() throws IOException {
+        if (spec.getLogConfiguration() != null) {
+            String confDir =
+                    createLogConfigFiles(
+                            spec.getLogConfiguration().get(CONFIG_FILE_LOG4J_NAME),
```

Comment:
Since only 2 specific file names (keys) will be recognized and used, we should validate that the user did not specify anything else. 

This check could go straight into the validator component.

Reply:
Thanks, @gyfora. Makes sense!

Reply:
@gyfora  Added the requested validation, PTAL.